File: data/code_pairs/cpp/598.cpp
#include <iostream>
#include <vector>

float f() {
    return 1.0f;
}

int main() {
    int a;
    int I;
    std::vector<bool> B(2);

    // Test 1: Test function f()
    if (f() == 1.0f) {
        std::cout << "Test 1 Passed\n";
    } else {
        std::cout << "Test 1 Failed\n";
    }
    
    // Setup for Test 2
    B = {true, false};
    if (std::any_of(B.begin(), B.end(), [](bool val) { return val; })) {
        a = 12;
    } else {
        a = 0;
    }
    if (a == 12) {
        std::cout << "Test 2 Passed\n";
    } else {
        std::cout << "Test 2 Failed\n";
    }
    
    // Setup for Test 3
    I = 1;
    if (I != 0) {
        a = 12;
    } else {
        a = 0;
    }
    if (a == 12) {
        std::cout << "Test 3 Passed\n";
    } else {
        std::cout << "Test 3 Failed\n";
    }
    
    return 0;
}

File: data/code_pairs/fortran/598.f90
module my_functions
  implicit none
contains
  real function f()
    f = 1.0
  end function f
end module my_functions

program main
  use my_functions
  implicit none
  integer :: a
  integer :: I
  logical, dimension(2) :: B
  real :: result

  ! Test 1: Test function f()
  result = f()
  if (result == 1.0) then
    print *, "Test 1 Passed"
  else
    print *, "Test 1 Failed"
  end if
  
  ! Setup for Test 2
  B = [ .true., .false. ]
  if (any(B)) then
    a = 12
  else
    a = 0
  end if
  if (a == 12) then
    print *, "Test 2 Passed"
  else
    print *, "Test 2 Failed"
  end if
  
  ! Setup for Test 3
  I = 1
  if (I /= 0) then
    a = 12
  else
    a = 0
  end if
  if (a == 12) then
    print *, "Test 3 Passed"
  else
    print *, "Test 3 Failed"
  end if
end program main